src/vm/clrtocomcall.cpp

```diff
@@ -1109,6 +1109,8 @@ void ComPlusMethodFrame::GetUnmanagedCallSite(TADDR* ip,
     }
     CONTRACTL_END;
 
+    _ASSERTE(false);
```

Comment:
This method is used by the debugger. You won't see it covered by the CoreCLR tests.

Reply:
Thanks, I wanted to see where we create `ComPlusMethodFrame` because `ComPlusMethodFrame::ScanRoots` calls `RetunsObject` and `ComPlusMethodFrame` doesn't have `EECodeInfo ` to get `ReturnKind` from.
https://github.com/dotnet/coreclr/blob/5ad457b6c1562907e81936a67955c98cf4034857/src/vm/frames.cpp#L1425-L1442

Using of `ReturnsObject` on arm64 for 16 bytes struct with pointers fails with assert:
https://github.com/dotnet/coreclr/blob/bdb995987178231ba541f22143cb3cab56309daa/src/vm/method.cpp#L1275
and on x64 Unix it doesn't report pointers in such structs and we have a GC hole in `ComPlusMethodFrame::GcScanRoots`.

Looks like I need to save `EECodeInfo ` when we create `ComPlusMethodFrame`:
https://github.com/dotnet/coreclr/blob/bdb995987178231ba541f22143cb3cab56309daa/src/vm/clrtocomcall.cpp#L965-L966

Reply:
`ComPlusMethodFrame` does not have `EECodeInfo`. The code to promote the return object should stay as is in `ComPlusMethodFrame::GcScanRoots`.